### PR TITLE
[SYCL] Fix SYCL-CTS regression after https://github.com/intel/llvm/pull/15543

### PR DESCRIPTION
**Pull request by aelovikov-intel**
Apparently, we perform some hacks around constant address space that I wasn't aware of. Make sure new address cast helpers are called consistently with the previous behavior.

### CODE REVIEW

**Review by sarnex — APPROVED** [review]
devops lgtm

File: sycl/include/sycl/multi_ptr.hpp [hunk 1]
```diff
@@ -1263,6 +1285,7 @@ class __SYCL2020_DEPRECATED(
 
   static constexpr access::address_space address_space = Space;
 
+public:
```
Comment (sergey-semenov) [inline sycl/include/sycl/multi_ptr.hpp:1288]:
```suggestion
```
Looks redundant to me.

**Review by sergey-semenov — APPROVED** [review]
Tiny nitpick aside, LGTM.

**Comment by KornevNikita** [comment]
@aelovikov-intel hi, it looks like the test fails after turning on

**Comment by aelovikov-intel** [comment]
> @aelovikov-intel hi, it looks like the test fails after turning on

I verified it locally when developing this fix and it passed. Will look into the failue.

**Comment by aelovikov-intel** [comment]
> > @aelovikov-intel hi, it looks like the test fails after turning on
> 
> I verified it locally when developing this fix and it passed. Will look into the failue.

The regression that I fixed here was resulting in a segmentation fault and that's been fixed. The current failures are of these form:

```
  generic_ptr alias. Core types
  -------------------------------------------------------------------------------
  /__w/llvm/llvm/khronos_sycl_cts/tests/multi_ptr/multi_ptr_explicit_conversions_core.cpp:41
  ...............................................................................
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/multi_ptr/multi_ptr_explicit_conversions_core.cpp:41: FAILED:
  explicitly with message:
    This test case has been compile-time disabled.
  
  -------------------------------------------------------------------------------
  generic_ptr alias. fp16 type
  -------------------------------------------------------------------------------
  /__w/llvm/llvm/khronos_sycl_cts/tests/multi_ptr/multi_ptr_explicit_conversions_fp16.cpp:45
  ...............................................................................
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/multi_ptr/multi_ptr_explicit_conversions_fp16.cpp:45: FAILED:
  explicitly with message:
    This test case has been compile-time disabled.
  
  -------------------------------------------------------------------------------
  generic_ptr alias. fp64 type
  -------------------------------------------------------------------------------
  /__w/llvm/llvm/khronos_sycl_cts/tests/multi_ptr/multi_ptr_explicit_conversions_fp64.cpp:45
  ...............................................................................
  
  /__w/llvm/llvm/khronos_sycl_cts/tests/multi_ptr/multi_ptr_explicit_conversions_fp64.cpp:45: FAILED:
  explicitly with message:
    This test case has been compile-time disabled.
  
  ===============================================================================
  test cases:   47 |   44 passed | 3 failed
  assertions: 3589 | 3586 passed | 3 failed
```

and I don't see how it can be related to the regression I had introduced and subsequently fixed by this PR.

**Comment by aelovikov-intel** [comment]
It looks like it's failing on the code added in https://github.com/KhronosGroup/SYCL-CTS/pull/935. Tagging @victor-eds to take a look into it.

**Pull request merged**